portfolio/src/main/webapp/script.js

```diff
@@ -31,16 +31,77 @@ function addRandomGreeting() {
  * Opens a modal with the given image and caption.
  */
 function openSimpleImageModal(modalId, modalImgId, captionId, closeId, imgSrc, captionText) {  
-  var modal = document.getElementById(modalId);
-  var modalImg = document.getElementById(modalImgId);
-  var caption = document.getElementById(captionId);
+  const modal = document.getElementById(modalId);
+  const modalImg = document.getElementById(modalImgId);
+  const caption = document.getElementById(captionId);
 
   modal.style.display = "block";
   modalImg.src = imgSrc;
-  caption.innerHTML = captionText;
+  caption.innerText = captionText;
 
-  var closeBtn = document.getElementById(closeId);
+  const closeBtn = document.getElementById(closeId);
   closeBtn.onclick = function() {
     modal.style.display = "none";
   };
 }
+
+/**
+ * Advance or Reverse the slide currently showing.
+ */
+function advanceSlides(n) {
+  currIndex += n;
+  currIndex = showSlidePicture(currIndex);
+};
+
+/**
+ * Show the image from the gallery at the given position.
+ */
+function showSlidePicture(n) {
```

Comment:
This function is probably complicated enough to have a unit test

Reply:
Added Jasmine using Karma to project, using Node. Added unit tests for the index calculation of this function (refactored first).